gateways/event-sources.go

```diff
@@ -246,8 +262,8 @@ func (gc *GatewayConfig) stopEventSources(configs []string) {
 }
 
 // manageEventSources syncs registered event sources and updated gateway configmap
-func (gc *GatewayConfig) manageEventSources(cm *corev1.ConfigMap) error {
-	eventSources, err := gc.createInternalEventSources(cm)
+func (gc *GatewayConfig) manageEventSources(es *esv1alpha1.EventSource) error {
```

Comment:
so from my understanding, CRDs are meant to represent infrastructure on k8s in the form of more primitive APIs.. it seems like this `EventSource` only has relevance/context when paired with a `Gateway`. Each event source doesn't create system state via resources, it does so inside of the gateway implementation, thus introducing the possibility that desired state != declared state. What about if the event sources themselves were actually pods? i think we can take a note from Knative here on this one.. check out [this](https://github.com/knative/docs/tree/master/docs/eventing/sources) and pay special attention to the [container event sources](https://github.com/knative/eventing-sources/blob/master/pkg/apis/sources/v1alpha1/containersource_types.go). They use the concept of a `sink` inside the CRD to refer a some object that resolves to a domain name to use as the target for the cloudEvents client.

Reply:
also following the above paradigm allows us to outsource the various event sources out of tree! in this implementation PR, you can't do that since they are defined as event source types in the API!